SE-XXX Typed throws.md

```diff
@@ -282,24 +303,113 @@ func userResultFromStrings(strings: [String]) throws GenericError -> User  {
 
 ```
 
-### Multiple layers of error flow
-
-`// TODO: Add examples and explain`
-
 ### Objective-C behaviour
 
+`// TODO: Not sure if we need that much history :D`
+
 In Objective-C, all current functions that take a double-pointer to `NSError` (a typical pattern in Foundation APIs) have an implicit type in the function signature, and, as has been pointed out, this does not provide more information that the current generic `Error`. But developers were free to subclass `NSError` and add it to its methods knowing that, the client would know at call time which kind of error would be raised if something went wrong.
 
+`// TODO: we should not pretend what is correct or the right way, there are always pros and cons, we should just summarize them`

```

Comment:
Here I just want to remark that even Swift ecosystem is starting to type its throwing functions even though they don't have typed `throws`

Reply:
that's a good point, I will note that in the todo